Homework2/Task2TrainingManagementSystem/Training.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace Task2TrainingManagementSystem
+{
+    class Training : Entities, ICloneable
+    {
+        
+        public List<Entities> Trainingslist = new List<Entities>();
+        
+        public Training(string description) : base(description)
+        {
+            Description = description;
+        }
+
+        public void Add(Entities element)
+        {
+            this.Trainingslist.Add(element);
+        }
+
+        public static bool IsPractical(List<Entities> Testinglist)
+        {
+            foreach(object element in Testinglist)
+            {
+                if(element is Lecture)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        public object Clone()
+        {
+            List<Entities> CopyList = new List<Entities>();
+
+            foreach(var item in Trainingslist)
+            {
+                CopyList.Add(item);
```

Comment:
This is not a clone. item is still original element. 
The code should clone elements of Trainingslist, each one. 

You can test Clone as following:

var originalTraining = new Training("desc1");
Lecture lecture = new Lecture("First lecture", "C#");
originalTraining.Add(lecture);

var clonedTraining = (Training)originalTraining.Clone();
lecture.Topic = "The updated topic!";
var clonedLecture =  (Lecture)clonedTraining.Trainingslist[0];

Console.WriteLine(clonedLecture.Topic); // I do believe you will see  "The updated topic!" here, meaning clonedLecture references to lecture. 
Console.WriteLine(clonedLecture == lecture ); //True should be here as clonedLecture references lecture.